lib/osi-licenses.yml

```
@@ -0,0 +1,54 @@
+# Generated by ChatGPT on 2023-11-18
+# https://chat.openai.com/c/9785e91f-37ab-4d13-8221-8e1bb10e8bc1
+
+Apache-2.0:
+  name: Apache License 2.0
+  url: https://opensource.org/licenses/Apache-2.0
+
+BSD-2-Clause:
+  name: BSD 2-Clause License
+  url: https://opensource.org/licenses/BSD-2-Clause
+
+BSD-3-Clause:
+  name: BSD 3-Clause License
+  url: https://opensource.org/licenses/BSD-3-Clause
+
+CDDL-1.0:
+  name: Common Development and Distribution License 1.0
+  url: https://opensource.org/licenses/CDDL-1.0
+
+EPL-2.0:
+  name: Eclipse Public License 2.0
+  url: https://opensource.org/licenses/EPL-2.0
+
+GPL-2.0:
+  name: GNU General Public License 2.0
+  url: https://opensource.org/licenses/GPL-2.0
+
+GPL-3.0-only:
+  name: GNU General Public License 3.0-only
+  url: https://opensource.org/licenses/GPL-3.0
+
+LGPL-2.0-only:
+  name: GNU Lesser General Public License 2.0-only
+  url: https://opensource.org/licenses/LGPL-2.0
+
+LGPL-2.1:
+  name: GNU Lesser General Public License 2.1
+  url: https://opensource.org/licenses/LGPL-2.1
+
+LGPL-3.0-only:
+  name: GNU Lesser General Public License 3.0-only
+  url: https://opensource.org/licenses/LGPL-3.0
+
+MIT:
+  name: MIT License
+  url: https://opensource.org/licenses/MIT
+
+MPL-2.0:
+  name: Mozilla Public License 2.0
+  url: https://opensource.org/licenses/MPL-2.0
+
+OTHER:
+  name: Other/uncommon license
+  url: https://opensource.org/licenses/?categories=popular-strong-community
```

```suggestion
```

I know ChatGPT came up with this list, but `OTHER` is out of place here. URL is definitely not to some sort of definition about what constitutes an `OTHER` license. Recommend removing.

---

I think it's critical for this list to have a set of principles, one of which is to use OSI's definition of common licenses.  I didn't want to add every niche license.

Unfortunately, when I added VV I learned that it uses the Affero license. That necessited an OTHER category, which I added manually.

---

@ahwagner : My preference is what's listed here. However, here are two other options:

1) Keep the OTHER label, but make the link point to the primary_url (i.e., the user needs to go figure it out)
2) Keep the OTHER label, but don't link at all

If you like one of those more than what's there, I'm happy to make the change.

---

Okay, I think that's fine. I took another look, and this can be easily resolved by updating to the correct URL (which does exist, but the original URL does not point there).

```suggestion
OTHER:
  name: Other/uncommon license
  url: https://opensource.org/licenses/?categories=other-miscellaneous
```

---

That's not actually the correct URL. OTHER in license.yml means that it's not one of the licenses that we choose to recognize. other-miscellaneous refers to licenses that don't fit the categories at OSI.

Example: Affero is used by VV, uncategorized at OSI, and listed as OTHER for us. Linking to other-miscellaneous is not correct.